typing/typemod.ml

```diff
@@ -2076,7 +2076,7 @@ let check_recmodule_inclusion env bindings =
         let coercion, shape =
           try
             Includemod.modtypes_with_shape ~shape
-              ~loc:modl.mod_loc ~mark:true
+              ~loc:modl.mod_loc ~mark:true ~mark_both:false
```

Comment:
This should be `~mark_both:true` too, otherwise the "recursive" version of the test raises an unused value warning:
```ocaml
module rec X: sig
   module F(_:sig val x:int end): sig end
end = struct
   module F(X:sig val x:int end) = struct let _ = X.x end
end
and Y: sig end = struct end
```

>```
>2 |    module F(_:sig val x:int end): sig end
>                      ^^^^^^^^^
>Warning 32 [unused-value-declaration]: unused value x.
>```